Makefile

```diff
@@ -139,14 +142,19 @@ clean:
 	rm -f node_modules/.uptodate
 	rm -rf build
 
-DOCKER_TAG = dev
-
-GULP := node_modules/.bin/gulp
-
 build/manifest.json: node_modules/.uptodate
 	$(GULP) build
 
 node_modules/.uptodate: package.json
 	@echo installing javascript dependencies
 	@node_modules/.bin/check-dependencies 2>/dev/null || npm install
 	@touch $@
+
+${PYTHON}:
+	pyenv install ${PYTHON_VERSION}
+	pip install --upgrade pip
+
+${TOX}: ${PYTHON}
+	pip install tox
+
+.IGNORE: ${PYTHON} ${TOX}
```

Comment:
This tells make to ignore failures when installing Python 3.6 and tox and continue anyway. If a developer has installed Python 3.6 and tox by some other means than pyenv, or if we're running on CI where they're installed in another way, calling pyenv will fail but make will continue and the `tox` command will still succeed.

Make does print out a warning, though, which creates noise in the logs, and I don't see a way to disable the warning. For that reason a better solution here might be a simple script that checks for `python3.6` and `tox` on `$PATH` and tries to use pyenv install them if they're not already on the `$PATH`.

If pyenv fails and Python 3.6 and tox have _not_ been installed by some other means then make will continue to the `tox` command and that will fail with some "Python 3.6 is not installed" or "tox is not installed" error. For example:

        $ make shell
        pyenv install 3.6.8
        make: pyenv: Command not found
        make: [Makefile:154: /home/seanh/.pyenv/versions/3.6.8] Error 127 (ignored)
        pip install --upgrade pip
        pyenv: version `3.6.8' is not installed (set by /home/seanh/Projects/h/.python-version)
        make: [Makefile:155: /home/seanh/.pyenv/versions/3.6.8] Error 1 (ignored)
        pip install tox
        pyenv: version `3.6.8' is not installed (set by /home/seanh/Projects/h/.python-version)
        make: [Makefile:158: /home/seanh/.pyenv/versions/3.6.8/bin/tox] Error 1 (ignored)
        tox -q -e py36-dev -- sh bin/hypothesis --dev shell
        pyenv: version `3.6.8' is not installed (set by /home/seanh/Projects/h/.python-version)
        make: *** [Makefile:47: shell] Error 1